### PR TITLE
chore(manager/gomod): Update go version to 1.23.3 in tests/examples

### PR DESCRIPTION
**Pull request by echarrod**
## Changes

- Update Go version to 1.23

## Context

- Go's [release policy](https://go.dev/doc/devel/release) only officially support the latest two major versions, i.e. 1.23 and 1.22 at the moment
- Originally was removing redundant flag in https://github.com/renovatebot/renovate/pull/32506, but wanted to ensure this wouldn't have any adverse effects for Go versions < 1.18

## Documentation (please check one with an [x])

- [X] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [X] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — COMMENTED** [review]
None of these changes makes any difference to how Renovate behaves at runtime

**Comment by echarrod** [comment]
> None of these changes makes any difference to how Renovate behaves at runtime

I just thought it would be more familiar for users to see an up to date Go version in the docs, that they would be using, and thought we might as well run the tests against a later Go version specified

**Comment by rarkins** [comment]
OK, I've updated this from `feat` to `chore`

**Review by rarkins — APPROVED** [review]

File: lib/modules/manager/gomod/__fixtures__/3/go-mod [hunk 1]
```diff
@@ -133,4 +133,3 @@ replace (
 )
 
 toolchain go1.22.1
```
Comment (viceice) [inline lib/modules/manager/gomod/__fixtures__/3/go-mod:135]: this should be removed, ho would remove it. should never lower than go directive

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/modules/manager/gomod/__fixtures__/3/go-mod (hunk 1 above)
Comment (echarrod) [reply lib/modules/manager/gomod/__fixtures__/3/go-mod:135]: Updated in https://github.com/renovatebot/renovate/pull/32546

**Pull request merged**